filewriter_v2/README.md

```diff
@@ -0,0 +1,390 @@
+| [pixel_mask_applied](https://manual.nexusformat.org/classes/applications/NXmx.html#nxmx-entry-instrument-detector-pixel-mask-applied-field) | NX_BOOLEAN | Indicates whether a pixel mask has been applied. |
+| [saturation_value](https://manual.nexusformat.org/classes/applications/NXmx.html#nxmx-entry-instrument-detector-saturation-value-field) | NX_INT | The value at which the detector goes into saturation. Data above this value is known to be invalid. |
+| [sensor_material](https://manual.nexusformat.org/classes/applications/NXmx.html#nxmx-entry-instrument-detector-sensor-material-field) | NX_CHAR | Material used for direct detection of X-rays in the sensor. |
+| [sensor_thickness](https://manual.nexusformat.org/classes/applications/NXmx.html#nxmx-entry-instrument-detector-sensor-thickness-field) | NX_FLOAT | Thickness of the sensor material. |
+| sensor_thickness@units | NX_LENGTH | `"m"` |
+| [serial_number](https://manual.nexusformat.org/classes/base_classes/NXdetector.html#nxdetector-serial-number-field) | NX_CHAR | Serial number of the detector. |
+| [start_time](https://manual.nexusformat.org/classes/base_classes/NXdetector.html#nxdetector-start-time-field) | NX_FLOAT[nP] | Relative start time for each frame, with `/entry/start_time` as absolute reference. |
+| [virtual_pixel_interpolation_applied](https://manual.nexusformat.org/classes/applications/NXmx.html#nxmx-entry-instrument-detector-virtual-pixel-interpolation-applied-field) | NX_BOOLEAN | Indicates whether virtual pixel interpolation has been applied. |
+| [x_pixel_size](https://manual.nexusformat.org/classes/base_classes/NXdetector.html#nxdetector-x-pixel-size-field) | NX_FLOAT | Size of a single pixel along the x-axis of the detector. |
+| x_pixel_size@units | NX_LENGTH | `"m"` |
+| [y_pixel_size](https://manual.nexusformat.org/classes/base_classes/NXdetector.html#nxdetector-y-pixel-size-field) | NX_FLOAT | Size of a single pixel along the y-axis of the detector. |
+| y_pixel_size@units | NX_LENGTH | `"m"` |
+
+
+## /entry/instrument/detector/CHANNELNAME_channel
+
+For every channel enabled, there will be one _NXdetector_channel_ group containing metadata specific to that channel. For further details see [Multi-Channel Data](#multi-channel-data).
+
+| Name | NeXus Type | Description |
+|---|---|---|
+| @NX_class | NX_CHAR | `"NXdetector_channel"` |
+| flatfield | NX_FLOAT[] | The flatfield applied to the channel data. |
+| pixel_mask | NX_UINT[] | The pixel mask applied to the channel data. |
+| threshold_energy | NX_FLOAT or NX_FLOAT[2] | The threshold energy for this channel. In case of difference mode, this is an array containing both the upper and lower threshold energies. |
+| threshold_energy@units | NX_ENERGY | `"eV"` |
+
+## [/entry/instrument/detector/geometry](https://manual.nexusformat.org/classes/applications/NXmx.html#nxmx-entry-instrument-detector-transformations-group)
```

Comment:
Why do we need both "geometry" and "transformations" groups? Should they be combined (e.g. only have "transformations")? Also, if we do want to keep goniometer parameters separate, we could also consider naming such a group "goniometer"?

Reply:
This is definitely subject to change. It should be discussed in a meeting Diego is organizing. But I agree, now that "geometry" is also of type "NXtransformations", we might group them together or name them differently.